Reenable Orion Cycling Support

DavidHuber-NOAA opened the pull request:
# Description
This updates the model hash to include the UPP update needed to be able to run the post processor on Orion, thus reenabling support on that system.

A note on the UPP: it is using a newer version of g2tmpl that requires a separate spack-stack 1.6.0 installation.  This version of g2tmpl will be standard in spack-stack 1.8.0, but for now requires loading separate modules for the UPP.

A note on running analyses on Orion: due to a yet-unknown issue causing the BUFR library to run much slower on Orion when compared with Rocky 8, the GSI and GDASApp are expected to run significantly slower than on any other platform (on the order of an hour longer).

Lastly, I made adjustments to the build_all.sh script to send more cores to compiling the UFS and GDASApp.  Under this configuration, the GSI, UPP, UFS_Utils, and WW3 pre/post executables finish compiling before the UFS when run with 20 cores.

Resolves #2694 
Resolves #2851 

# Type of change
- Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO
- Does this change require an update to any of the following submodules? YES (If YES, please add a link to any PRs that are pending.)
  - [ ] EMC verif-global
  - [ ] GDAS
  - [ ] GFS-utils
  - [ ] GSI
  - [ ] GSI-monitor
  - [ ] GSI-utils
  - [ ] UFS-utils
  - [x] UFS-weather-model
  - [ ] wxflow

# How has this been tested?
- [x] Build on Orion
- [x] C48_ATM on Orion
- [x] C48_S2SW on Orion
- [x] C96C48_hybatmDA on Orion

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] New and existing tests pass with my changes

DavidHuber-NOAA committed:
Update model hash, reenable Orion cycling support

DavidHuber-NOAA committed:
Adjust walltime for eobs, rename walltime variables

WalterKolczynski-NOAA commented on sorc/link_workflow.sh:
```diff
@@ -162,15 +162,22 @@ cd "${HOMEgfs}/parm/ufs" || exit 1
 ${LINK_OR_COPY} "${HOMEgfs}/sorc/ufs_model.fd/tests/parm/noahmptable.tbl" .
 
 cd "${HOMEgfs}/parm/post" || exit 1
```
I think we should follow the UPP and put these into separate `gfs` and `gefs` subdirectories now.

DavidHuber-NOAA replied:
OK, fixed.

DavidHuber-NOAA committed:
Link gfs and gefs upp parm directories instead of files

DavidHuber-NOAA committed:
Load more modules for the UPP

WalterKolczynski-NOAA commented on jobs/rocoto/upp.sh:
```diff
@@ -36,7 +36,30 @@ if [[ "${MACHINE_ID}" = "wcoss2" ]]; then
 
 else
-  . "${HOMEgfs}/ush/load_fv3gfs_modules.sh"
+  # For spack-stack 1.6.0, the UPP has its own environment (upp-addon-dev)
+  source "${HOMEgfs}/ush/module-setup.sh"
+  module use "${HOMEgfs}/sorc/ufs_model.fd/FV3/upp/modulefiles"
+  module load "${MACHINE_ID}"
+  source "${HOMEgfs}/versions/run.spack.ver"
+  module load "prod_util/${prod_util_ver}"
+  module load "grib-util/${grib_util_ver}"
+  # The numpy version in the UPP environment differs from the gsi-addon
+  module load "py-numpy/1.22.3"
+  module load "py-pyyaml/${py_pyyaml_ver}"
+  module load "py-netcdf4/${py_netcdf4_ver}"
+  module load "py-f90nml/${py_f90nml_ver}"
+
+  # The jinja2 version in the UPP environment differs from the gsi-addon
+  module load "py-jinja2/3.0.3"
+
+  # xarray is not in this environment...
+  # module load "py-xarray/${py_xarray_ver}"
+
+  wxflowPATH="${HOMEgfs}/ush/python"
+  PYTHONPATH="${PYTHONPATH:+${PYTHONPATH}:}${HOMEgfs}/ush:${wxflowPATH}"
+  export PYTHONPATH
+  #. "${HOMEgfs}/ush/load_fv3gfs_modules.sh"
```
Can we just use the UFS module here?

DavidHuber-NOAA replied:
Unfortunately not.  The UPP requires crtm/2.4.0.1, g2tmpl/1.13.0, and g2/3.5.1, all of which are newer than what is in the UFS (unified-dev) environment.

DavidHuber-NOAA replied:
On another look, it appears that we can since the libraries are statically linked at build time.  I will run a test to verify that there are no changes to the master grib2 files when running with the fv3gfs modules vs the UPP modules.

DavidHuber-NOAA replied:
The master grib2 files were bitwise identical when using module_base.orion or the UPP's `orion` module.

DavidHuber-NOAA commented:
@WenMeng-NOAA

The spack-stack environment used by the UPP (`upp-addon`) does not include `xarray` and uses older versions of `numpy` (v1.22.3) and `jinja2` (3.0.3) than the `gsi-addon` environment.  To make this work, I had to add a kludge to `ush/python/pygfs/__init.py__` so that the marine `Task`s' importing of `xarray` does not cause the UPP to fail.  This can be removed once we get to spack-stack v1.8.0.

With these changes, the `gdasatmanlupp` and `gfsatmanlupp` jobs ran to completion.  Wen, would you be able to look at the output files and verify they are OK?  You can find a pair of examples here:
```
/work/noaa/global/dhuber/para/orion/COMROOT/c96_4denvar/gdas.20211221/00/model/atmos/master/gdas.t00z.master.grb2anl
/work/noaa/global/dhuber/para/orion/COMROOT/c96_4denvar/gfs.20211221/00/model/atmos/master/gfs.t00z.master.grb2anl
```

DavidHuber-NOAA committed:
Merge branch 'develop' into feature/orion_upp_update

WenMeng-NOAA commented:
@DavidHuber-NOAA Can you change the access permission of /work/noaa/global/dhuber/para/orion/COMROOT/c96_4denvar/gdas.20211221/00/model/atmos/master/?

DavidHuber-NOAA commented:
@WenMeng-NOAA Yes, done.

DavidHuber-NOAA committed:
Update path to post_tag_gfs*

DavidHuber-NOAA committed:
Merge branch 'feature/orion_upp_update' of github.com:davidhuber-noaa/global-workflow into feature/orion_upp_update

DavidHuber-NOAA committed:
Update UPP parm paths

DavidHuber-NOAA committed:
Revert UPP modules

DavidHuber-NOAA committed:
Remove extra whitespace

WenMeng-NOAA commented on sorc/link_workflow.sh:
```diff
@@ -162,15 +162,14 @@ cd "${HOMEgfs}/parm/ufs" || exit 1
-    postxconfig-NT-GFS-ANL.txt postxconfig-NT-GFS-F00.txt postxconfig-NT-GFS-FLUX-F00.txt \
-    postxconfig-NT-GFS.txt postxconfig-NT-GFS-FLUX.txt postxconfig-NT-GFS-GOES.txt \
-    postxconfig-NT-GFS-F00-TWO.txt postxconfig-NT-GFS-TWO.txt \
-    params_grib2_tbl_new post_tag_gfs128 post_tag_gfs65 nam_micro_lookup.dat
```
@DavidHuber-NOAA post_tag_gfs128 and post_tag_gfs65 should be linked to parm/post instead of parm/post/gfs.

DavidHuber-NOAA replied:
Alright, fixed.

aerorahul replied:
They are not used in the global-workflowl

WenMeng-NOAA replied:
@aerorahul Could you let me know location of the 'itag' namelist for inline post?

aerorahul replied:
@WenMeng-NOAA @DavidHuber-NOAA 
I stand corrected.  We use the `post_tag_gfs128` file for inline post.
@WenMeng-NOAA 
Will there be a `post_tag_gefs128` file for inline post for GEFS?

@WalterKolczynski-NOAA
What should we do? Maintain our copy or use the UPP provided `post_tag_gfsXXX`?

aerorahul replied:
@WenMeng-NOAA The global-workflow currently uses https://github.com/NOAA-EMC/UPP/blob/develop/parm/gfs/post_tag_gfs128 for inline post
It does not have `rdaod=.true.`

WenMeng-NOAA replied:
@aerorahul @WalterKolczynski-NOAA Could these 'itag" namelists for gfs/gefs be saved in the global-workflow? It would be flexible.

WenMeng-NOAA commented:
> @WenMeng-NOAA
> 
> The spack-stack environment used by the UPP (`upp-addon`) does not include `xarray` and uses older versions of `numpy` (v1.22.3) and `jinja2` (3.0.3) than the `gsi-addon` environment. To make this work, I had to add a kludge to `ush/python/pygfs/__init.py__` so that the marine `Task`s' importing of `xarray` does not cause the UPP to fail. This can be removed once we get to spack-stack v1.8.0.
> 
> With these changes, the `gdasatmanlupp` and `gfsatmanlupp` jobs ran to completion. Wen, would you be able to look at the output files and verify they are OK? You can find a pair of examples here:
> 
> ```
> /work/noaa/global/dhuber/para/orion/COMROOT/c96_4denvar/gdas.20211221/00/model/atmos/master/gdas.t00z.master.grb2anl
> /work/noaa/global/dhuber/para/orion/COMROOT/c96_4denvar/gfs.20211221/00/model/atmos/master/gfs.t00z.master.grb2anl
> ```

@DavidHuber-NOAA Somehow, the 6 aerosol fields (ATOK) are missing in both gfs and gdas master files. To output them from UPP, itag should be set as:
```
&nampgb
  kpo = 57,
  po = 1000.0, 975.0, 950.0, 925.0, 900.0, 875.0, 850.0, 825.0, 800.0, 775.0, 750.0, 725.0, 700.0, 675.0, 650.0, 625.0, 600.0, 575.0, 550.0, 525.0, 500.0, 475.0, 450.0, 425.0, 400.0, 375.0, 350.0, 325.0, 300.0, 275.0, 250.0, 225.0, 200.0, 175.0, 150.0, 125.0, 100.0, 70.0, 50.0, 40.0, 30.0, 20.0, 15.0, 10.0, 7.0, 5.0, 3.0, 2.0, 1.0, 0.7, 0.4, 0.2, 0.1, 0.07, 0.04, 0.02, 0.01,
  rdaod = .true.
```

DavidHuber-NOAA commented:
@WenMeng-NOAA Here are the contents of the `nampgb` namelist used to generate the master file:

```
&nampgb
  kpo = 57,
  po = 1000.0, 975.0, 950.0, 925.0, 900.0, 875.0, 850.0, 825.0, 800.0, 775.0, 750.0, 725.0, 700.0, 675.0, 650.0, 625.0, 600.0, 575.0, 550.0, 525.0, 500.0, 475.0, 450.0, 425.0, 400.0, 375.0, 350.0, 325.0, 300.0, 275.0, 250.0, 225.0, 200.0, 175.0, 150.0, 125.0, 100.0, 70.0, 50.0, 40.0, 30.0, 20.0, 15.0, 10.0, 7.0, 5.0, 3.0, 2.0, 1.0, 0.7, 0.4, 0.2, 0.1, 0.07, 0.04, 0.02, 0.01,
  rdaod = .true.
```

These appear to be the same.  The experiment that I was running was ATM-only, so that may be why the aerosol fields were not present.  I will try running the `C96C48_hybatmaerosnowDA` test case and let you know when I have an analysis master grib2 file ready.

DavidHuber-NOAA committed:
Link post_tag files to parm/post

DavidHuber-NOAA committed:
Merge remote-tracking branch 'origin/develop' into feature/orion_upp_update

DavidHuber-NOAA commented:
@WenMeng-NOAA I have finished running the `C96C48_hybatmaerosnowDA` test case.  The analysis grib2 file can be found here: `/work/noaa/global/dhuber/para/orion/COMROOT/hybaerosnow/gdas.20211221/00/model/atmos/master/gdas.t00z.master.grb2anl`.  Running `wgrib2 | grep ATOK` returned no entries.  Is this a feature that requires a new g2 or g2tmpl module?  Or is there perhaps a change in another namelist elsewhere?

WenMeng-NOAA commented:
> @WenMeng-NOAA I have finished running the `C96C48_hybatmaerosnowDA` test case. The analysis grib2 file can be found here: `/work/noaa/global/dhuber/para/orion/COMROOT/hybaerosnow/gdas.20211221/00/model/atmos/master/gdas.t00z.master.grb2anl`. Running `wgrib2 | grep ATOK` returned no entries. Is this a feature that requires a new g2 or g2tmpl module? Or is there perhaps a change in another namelist elsewhere?

@DavidHuber-NOAA There are no new g2/g2tmpl requirement. It's most likely  a namelist issue. Could you point me to the runtime logs?

CoryMartin-NOAA commented:
@WenMeng-NOAA I think this is related to what I emailed you about a few weeks ago. I saw similar behavior in testing on Hera from a previous version of the workflow, so I don't think this is anything unique to Orion or this PR. I think the issue is that the workflow default namelists are not producing `AOTK` in the products when the aerosol component is turned on, but the UPP control file you provided me works perfectly

WenMeng-NOAA commented:
> @WenMeng-NOAA I think this is related to what I emailed you about a few weeks ago. I saw similar behavior in testing on Hera from a previous version of the workflow, so I don't think this is anything unique to Orion or this PR. I think the issue is that the workflow default namelists are not producing `AOTK` in the products when the aerosol component is turned on, but the UPP control file you provided me works perfectly

The AOTK is only included in the UPP control file for forecast hours, so we wouldn't expect 'AOTK' generated in master and pgrb2 files for analysis. 
To generate 'AOTK' from inline post for forecast hours,  the option "rdaod = .true." should be included in the namelist 'post_tag_gfs128' which is linked from UPP repository. We could address this in the next PR.

DavidHuber-NOAA commented:
The naming convention for the `gfsgoes` `upp` products have updated from `GFSPRS.GrbFxxx` to `GFSGOES.GrbFxxx`, which caused the WCOSS2 failures.  The `upp` `yaml` file needs to be updated to reflect this.

DavidHuber-NOAA committed:
Update gfsgoes file name convention

DavidHuber-NOAA commented:
@aerorahul On top of the new name for the `gfsgoes` `upp` output file, the `.idx` file was not generated.  When `WCOSS2` comes back up, would you mind taking a look at the job and seeing if anything else needs to be updated?

aerorahul commented on sorc/link_workflow.sh:
```diff
@@ -162,15 +162,19 @@ cd "${HOMEgfs}/parm/ufs" || exit 1
+  ${LINK_OR_COPY} "${HOMEgfs}/sorc/upp.fd/parm/${dir}" .
+done
+# Link the post_tag files into parm/post from upp.fd/parm/gfs
+for file in post_tag_gfs128 post_tag_gfs65
+do
+  ${LINK_OR_COPY} "${HOMEgfs}/sorc/upp.fd/parm/gfs/${file}" .
+done
```
We don't use these files in the workflow. They are hardwired for a test in the UPP (I think)

DavidHuber-NOAA replied:
We do use a few files from `parm/gfs`: `postxconfig-NT-gfs-anl.txt`, `postxconfig-NT-gfs-f00-two.txt`, and `postxconfig-NT-gfs-goes.txt`.

aerorahul replied:
Yes.  Those `txt` files are used as the `flatfile`. The `post_tag_*` files are the namelists that are hardwired with little info especially in the `model_inputs` namelist.
We will generate the namelist on the fly until UPP can allow improved flexibility and control over providing input/output files, etc.

DavidHuber-NOAA replied:
Oh, misread the lines...  Disregard.

aerorahul commented on .gitignore:
```diff
@@ -71,46 +68,14 @@ parm/post/optics_luts_WASO_nasa.dat
 parm/post/params_grib2_tbl_new
 parm/post/post_tag_gfs128
 parm/post/post_tag_gfs65
```
Should be removed.

aerorahul commented on parm/post/upp.yaml:
```diff
@@ -81,10 +81,10 @@ goes:
     copy:
-      - ["{{ DATA }}/GFSPRS.GrbF{{ '%02d' % forecast_hour }}", "{{ COM_ATMOS_MASTER }}/{{ RUN }}.t{{ current_cycle | strftime('%H') }}z.special.grb2f{{ '%03d' % forecast_hour }}"]
-      - ["{{ DATA }}/GFSPRS.GrbF{{ '%02d' % forecast_hour }}.idx", "{{ COM_ATMOS_MASTER }}/{{ RUN }}.t{{ current_cycle | strftime('%H') }}z.special.grb2if{{ '%03d' % forecast_hour }}"]
+      - ["{{ DATA }}/GFSGOES.GrbF{{ '%02d' % forecast_hour }}", "{{ COM_ATMOS_MASTER }}/{{ RUN }}.t{{ current_cycle | strftime('%H') }}z.special.grb2f{{ '%03d' % forecast_hour }}"]
```
@WenMeng-NOAA 
Is there a way to provide the name of the expected output file `GFSPRS.GrbFHH` or `GFSGOES.GrbFHH` via the namelist?  
It provides greater flexibility

aerorahul commented:
> @aerorahul On top of the new name for the `gfsgoes` `upp` output file, the `.idx` file was not generated. When `WCOSS2` comes back up, would you mind taking a look at the job and seeing if anything else needs to be updated?

Yes.  When WCOSS2 comes back after the switch, we will re-run wcoss2 test. The `idx` file is the index file based on the `GFSGEOS.FHH` file, so, it is expected to not have been created.

WenMeng-NOAA commented:
> The naming convention for the `gfsgoes` `upp` products have updated from `GFSPRS.GrbFxxx` to `GFSGOES.GrbFxxx`, which caused the WCOSS2 failures. The `upp` `yaml` file needs to be updated to reflect this.

The gfsgoes generation shouldn't be turned on from inline post in CI configuration.

aerorahul commented:
> > The naming convention for the `gfsgoes` `upp` products have updated from `GFSPRS.GrbFxxx` to `GFSGOES.GrbFxxx`, which caused the WCOSS2 failures. The `upp` `yaml` file needs to be updated to reflect this.
> 
> The gfsgoes generation shouldn't be turned on from inline post in CI configuration.

GFSGOES is not generated via inline post.  It is run via standalone execution of the UPP

aerorahul commented:
> /work2/noaa/stmp/CI/ORION/2877/RUNTESTS/COMROOT/C48_S2SWA_gefs_200cf26b/logs/2021032312/wave_post_grid_mem002.log

I am not sure why the gfs_wave_sbs jobs are failing in this PR.  There should be no changes to impact those.

DavidHuber-NOAA committed:
Remove 65-level upp itag files.

WenMeng-NOAA commented:
> > > The naming convention for the `gfsgoes` `upp` products have updated from `GFSPRS.GrbFxxx` to `GFSGOES.GrbFxxx`, which caused the WCOSS2 failures. The `upp` `yaml` file needs to be updated to reflect this.
> > 
> > 
> > The gfsgoes generation shouldn't be turned on from inline post in CI configuration.
> 
> GFSGOES is not generated via inline post. It is run via standalone execution of the UPP

Does this test only turn on for WCOSS2, or other platforms? It usually costs more time to complete goes generation.

WalterKolczynski-NOAA commented:
> > > > The naming convention for the `gfsgoes` `upp` products have updated from `GFSPRS.GrbFxxx` to `GFSGOES.GrbFxxx`, which caused the WCOSS2 failures. The `upp` `yaml` file needs to be updated to reflect this.
> > > 
> > > 
> > > The gfsgoes generation shouldn't be turned on from inline post in CI configuration.
> > 
> > 
> > GFSGOES is not generated via inline post. It is run via standalone execution of the UPP
> 
> Does this test only turn on for WCOSS2, or other platforms? It usually costs more time to complete goes generation.

Just WCOSS

aerorahul commented on sorc/link_workflow.sh:
```diff
@@ -162,15 +162,16 @@ cd "${HOMEgfs}/parm/ufs" || exit 1
+do
+  ${LINK_OR_COPY} "${HOMEgfs}/sorc/upp.fd/parm/${dir}" .
+done
+# Link the 128-level post_tag file into parm/post from upp.fd/parm/gfs
+${LINK_OR_COPY} "${HOMEgfs}/sorc/upp.fd/parm/gfs/post_tag_gfs128" .
```
```suggestion
```
Not used in the workflow

aerorahul committed:
Update sorc/link_workflow.sh

aerorahul commented on .gitignore:
```diff
@@ -70,47 +67,14 @@ parm/post/optics_luts_WASO.dat
 parm/post/optics_luts_WASO_nasa.dat
 parm/post/params_grib2_tbl_new
 parm/post/post_tag_gfs128
```
```suggestion
```

aerorahul committed:
Update .gitignore

WalterKolczynski-NOAA committed:
Wait to run wave post until after forecast completes

The new forecast segments capability introduced an issue where the
wave post job could time out waiting for the output of later segments
before those segments were complete. For now, the dependency is changed
to wait until the forecast is complete until wave post can be refactored.

DavidHuber-NOAA committed:
Merge pull request #14 from WalterKolczynski-NOAA/feature/orion_upp_update

Wait to run wave post until after forecast completes

WenMeng-NOAA commented:
> Experiment C48_ATM_59f3643f **FAIL** on Wcoss2 at 09/04/24 04:48:20 PM
> 
> Error logs:
> 
> ```
> /lfs/h2/emc/global/noscrub/globalworkflow.ci/GFS_CI_ROOT/PR/2877/RUNTESTS/COMROOT/C48_ATM_59f3643f/logs/2021032312/gfsfcst_seg0.log
> ```
> 
> Follow link here to view the contents of the above file(s): [(link)](https://gist.github.com/emcbot/03537eaa3f74e094848898902d80323d)
```
 forecast_predet.sh[540]: /bin/cp -p /lfs/h2/emc/global/noscrub/globalworkflow.ci/GFS_CI_ROOT/PR/2877/global-workflow/parm/post/post_tag_gfs128 /lfs/h2/emc/stmp/terry.mcguinness/RUNDIRS/C48_ATM_59f3643f/gfs.2021032312/gfsfcst.2021032312/fcst.98972/itag
/bin/cp: cannot stat '/lfs/h2/emc/global/noscrub/globalworkflow.ci/GFS_CI_ROOT/PR/2877/global-workflow/parm/post/post_tag_gfs128': No such file or directory
+ forecast_predet.sh[1]: postamble exglobal_forecast.sh 1725468022 1
```

aerorahul committed:
Merge branch 'develop' into feature/orion_upp_update

DavidHuber-NOAA committed:
Link in the 128-level itag file

DavidHuber-NOAA commented:
After speaking with @WenMeng-NOAA, I added the 128-level `itag` file back to its original location in parm/post.  Kicking off another set of tests on Orion and WCOSS2.

aerorahul committed:
fixes for UPP and compression in ufswm

aerorahul committed:
Merge branch 'feature/orion_upp_update' into patch/2877

aerorahul committed:
post tag name is incorrect

aerorahul committed:
fix shellnorm errors

aerorahul committed:
Update ush/forecast_predet.sh

Co-authored-by: David Huber <69919478+DavidHuber-NOAA@users.noreply.github.com>

aerorahul committed:
comment cleanup

aerorahul committed:
Merge branch 'patch/2877' of https://github.com/aerorahul/global-workflow into patch/2877

DavidHuber-NOAA committed:
Merge pull request #15 from aerorahul/patch/2877

fixes for UPP and compression in ufswm

WalterKolczynski-NOAA commented:
The index file isn't being created in the GOES UPP job. I checked `$DATA` and it isn't there (`GFSGOES.GrbF00` _is_ there).
```
[38;21m2024-09-05 22:30:49,031 - INFO     - upp         : Copy 'goes' processed data to COM/ directory[0m
[38;21m2024-09-05 22:30:49,034 - INFO     - file_utils  : Copied /lfs/h2/emc/stmp/terry.mcguinness/RUNDIRS/C96_atm3DVar_extended_c8710fe9/gfs.2021122100/upp.161988/GFSGOES.GrbF00 to /lfs/h2/emc/global/noscrub/globalworkflow.ci/GFS_CI_ROOT/PR/2877/RUNTESTS/COMROOT/C96_atm3DVar_extended_c8710fe9/gfs.20211221/00//model/atmos/master/gfs.t00z.special.grb2f000[0m
Traceback (most recent call last):
  File "/lfs/h2/emc/global/noscrub/globalworkflow.ci/GFS_CI_ROOT/PR/2877/global-workflow/ush/python/wxflow/fsutils.py", line 85, in cp
    shutil.copy2(source, target)
  File "/apps/spack/python/3.8.6/intel/19.1.3.304/pjn2nzkjvqgmjw4hmyz43v5x4jbxjzpk/lib/python3.8/shutil.py", line 432, in copy2
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/apps/spack/python/3.8.6/intel/19.1.3.304/pjn2nzkjvqgmjw4hmyz43v5x4jbxjzpk/lib/python3.8/shutil.py", line 261, in copyfile
    with open(src, 'rb') as fsrc, open(dst, 'wb') as fdst:
FileNotFoundError: [Errno 2] No such file or directory: '/lfs/h2/emc/stmp/terry.mcguinness/RUNDIRS/C96_atm3DVar_extended_c8710fe9/gfs.2021122100/upp.161988/GFSGOES.GrbF00.idx'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/lfs/h2/emc/global/noscrub/globalworkflow.ci/GFS_CI_ROOT/PR/2877/global-workflow/scripts/exglobal_atmos_upp.py", line 48, in <module>
    main()
  File "/lfs/h2/emc/global/noscrub/globalworkflow.ci/GFS_CI_ROOT/PR/2877/global-workflow/ush/python/wxflow/logger.py", line 266, in wrapper
    retval = func(*args, **kwargs)
  File "/lfs/h2/emc/global/noscrub/globalworkflow.ci/GFS_CI_ROOT/PR/2877/global-workflow/scripts/exglobal_atmos_upp.py", line 44, in main
    upp.finalize(upp_dict.upp_run, upp_yaml)
  File "/lfs/h2/emc/global/noscrub/globalworkflow.ci/GFS_CI_ROOT/PR/2877/global-workflow/ush/python/wxflow/logger.py", line 266, in wrapper
    retval = func(*args, **kwargs)
  File "/lfs/h2/emc/global/noscrub/globalworkflow.ci/GFS_CI_ROOT/PR/2877/global-workflow/ush/python/pygfs/task/upp.py", line 264, in finalize
    FileHandler(upp_yaml[upp_run].data_out).sync()
  File "/lfs/h2/emc/global/noscrub/globalworkflow.ci/GFS_CI_ROOT/PR/2877/global-workflow/ush/python/wxflow/file_utils.py", line 43, in sync
    sync_factory[action](files)
  File "/lfs/h2/emc/global/noscrub/globalworkflow.ci/GFS_CI_ROOT/PR/2877/global-workflow/ush/python/wxflow/file_utils.py", line 63, in _copy_files
    cp(src, dest)
  File "/lfs/h2/emc/global/noscrub/globalworkflow.ci/GFS_CI_ROOT/PR/2877/global-workflow/ush/python/wxflow/fsutils.py", line 87, in cp
    raise OSError(f"unable to copy {source} to {target}")
OSError: unable to copy /lfs/h2/emc/stmp/terry.mcguinness/RUNDIRS/C96_atm3DVar_extended_c8710fe9/gfs.2021122100/upp.161988/GFSGOES.GrbF00.idx to 
```

aerorahul commented:
Let's disable the GOES product generation and merge this PR.  
Something has changed unintentionally in there that needs to be further investigated.  
HR4 needs to commence immediately.

aerorahul commented:
https://github.com/NOAA-EMC/UPP/commit/97ea6555f855d136198738bd181ff52ca2ad0d31
here GFSGOES was renamed in the post_gfs_goes control file.

https://github.com/NOAA-EMC/global-workflow/blob/develop/ush/python/pygfs/task/upp.py#L203-L207
The index file is created from `GFSPRS` and `GFSFLX` files.  The above change in from `GFSPRS` to `GFSGOES` broke this part of the workflow.

aerorahul commented:
@DavidHuber-NOAA 
https://github.com/NOAA-EMC/global-workflow/blob/2e4f4b7671cfec83331ec26de39218543d7b9a6d/ush/python/pygfs/task/upp.py#L205
Can you add `GOES` to the list here?
`['PRS', 'FLX', 'GOES']`
This should be done properly, but this will do in a pinch.

DavidHuber-NOAA committed:
Add GOES to the list of indexable file types.
